src/java/org/apache/cassandra/db/streaming/ComponentManifest.java

```diff
@@ -106,6 +106,14 @@ public int hashCode()
         return components.hashCode();
     }
 
+    @Override
+    public String toString()
+    {
+        return "ComponentManifest{" +
+               "components=" + components +
+               '}';
```

Comment:
style: could also be `return String.format("ComponentManifest{components=%s}", components);` if you like to see the whole message format without substitutions.

Reply:
I rather not make this change.  String.format is super slow, so if we don't know the cases that call this then its not always safe to call.